Sources/WalletConnectSign/Engine/Common/DisconnectService.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+import WalletConnectPairing
+
+class DisconnectService {
+    enum Errors: Error {
+        case objectForTopicNotFound
+    }
+
+    private let deletePairingService: DeletePairingService
+    private let deleteSessionService: DeleteSessionService
+    private let pairingStorage: WCPairingStorage
+    private let sessionStorage: WCSessionStorage
+
+    init(deletePairingService: DeletePairingService,
+         deleteSessionService: DeleteSessionService,
+         pairingStorage: WCPairingStorage,
+         sessionStorage: WCSessionStorage) {
+        self.deletePairingService = deletePairingService
+        self.deleteSessionService = deleteSessionService
+        self.pairingStorage = pairingStorage
+        self.sessionStorage = sessionStorage
+    }
+
+    func disconnect(topic: String) async throws {
+        if pairingStorage.hasPairing(forTopic: topic) {
+            try await deletePairingService.delete(topic: topic)
+        } else if sessionStorage.hasSession(forTopic: topic) {
+            try await deleteSessionService.delete(topic: topic)
```

Comment:
Pairing and a session at the same time? I mean: do we need to clear pairing and session together on disconnect call?

Reply:
if so, I think is better to have 1 DeleteSessionService, which clean pairing and session storage

Reply:
one topic corresponds to one pairing or one session and we have a common public method for deletion.
So we either delete a pairing or a session.